packages/api/amplify_api/lib/src/graphql/ws/web_socket_connection.dart

```diff
@@ -110,14 +171,111 @@ class WebSocketConnection implements Closeable {
   @override
   void close([int closeStatus = _defaultCloseStatus]) {
     _logger.verbose('Closing web socket connection.');
+    _hubEventsController?.add(SubscriptionHubEvent.disconnected());
+    _rebroadcastController.close();
+
+    _resetConnectionInit();
+    _resetConnectionVariables(closeStatus);
+
+    _subscriptionRequests.clear();
+    _hubEventsController?.close();
+    _hubEventsController = null;
+  }
+
+  /// Initiate polling
+  void _startPolling() {
+    final interval = _subscriptionOptions?.pingInterval ?? _defaultPingInterval;
+
+    _pingTimer?.cancel();
+    _pingTimer = Timer.periodic(interval, (_) => _poll());
+    _poll();
+  }
+
+  /// Pings AppSync to verify health of connection. Failed pings trigger
+  /// reconnection through retry/back off.
+  Future<void> _poll() async {
+    try {
+      if (_pingClient == null) return;
+      final res = await _getPollRequest();
+      if (res!.body != 'healthy') {
+        throw Exception([
```

Comment:
Throwing here doesn't do much since you immediately catch it and discard the error. Maybe log the error in the catch block?

I would make this an `ApiException` as well vs a regular `Exception` and add a note about internet connection in the recoveryMessage.